internal/httprouter/config.go

```diff
@@ -0,0 +1,70 @@
+package router
+
+import (
+	"boilerplate/internal/framework"
+	"context"
+	"fmt"
+	"github.com/spf13/viper"
+	"go.uber.org/dig"
+	"go.uber.org/fx"
+	"go.uber.org/zap"
+	"net/http"
+)
+
+type ModuleConfig struct {
+	Port                   int `mapstructure:"HTTP_APP_PORT"`
+	RedirectTrailingSlash  bool
+	RedirectFixedPath      bool
+	HandleMethodNotAllowed bool
+	HandleOPTIONS          bool
+	NotFound               http.Handler
+	MethodNotAllowed       http.Handler
+	PanicHandler           func(http.ResponseWriter, *http.Request, interface{})
+	container              *dig.Container
+}
+
+func ModuleHooks(
+	lc fx.Lifecycle,
+	router *Router,
+	routes *framework.Routes,
+	logger *zap.Logger,
+) error {
+	lc.Append(
+		fx.Hook{
+			OnStart: func(context.Context) error {
+				router.AddRoutes(routes.GetRoutesInfo())
+				logger.Info(fmt.Sprintf("Listen to the port: %d", router.port))
+				go http.ListenAndServe(fmt.Sprintf(":%d", router.port), router.router)
+				return nil
+			},
```

Comment:
I propose to call all initialization code directly in functions which called by Provide/Invoke/etc and in hooks we should start/stop servers and maybe connect/disconnect to db. Here no difference in this case, but sometimes it can be useful, because of difference how fx calls invokes and hooks

Reply:
I am afraid, that it will break the ability to override some hosts. Imagine we have a common module auth, that has a route `/login`, but in my application, I want to override it to `v1/login`. In the case, of adding routes to the router inside the ModuleHooks function, I have to keep the strict order of modules, where the module of a router should go after the overriding module. 

In a case, of adding routes directly before running the router - it is no so tough coupling between modules.